services/wireguard/endpoint/endpoint.go

```diff
@@ -92,6 +100,18 @@ func (ce *connectionEndpoint) Start(config *wg.ServiceConfig) error {
 	return ce.wgClient.ConfigureDevice(ce.iface, deviceConfig, ce.ipAddr)
 }
 
+func (ce *connectionEndpoint) mapPort(port int) {
+	if ce.outboundIP != ce.publicIP {
+		ce.connectDelay = connectDelayForPortMap
```

Comment:
This looks like a hack, are we sure that 2000ms will be good for every router? Why 2000ms?

Reply:
2000 just by experimentation only. I cannot think of nice enough way to determine if router really activated this forwarding. Rule itself gets accepted, but looks like it is still not active for sometime. IDG protocol is too dum in this case and we don't have any status. 
I believe it might be heavy dependent on a gateway manufacturer how long does it take to activate such rule and we should perform tests with various models.
If you have some router differing from ours ( Mikrotik ) you can test this too.